src/transformers/generation_utils.py

```diff
@@ -1970,8 +1970,19 @@ def sample(
                         else (outputs.hidden_states,)
                     )
 
+            # To avoid all `-inf` along the vocab dimension (dim -1), which gives `nan` after `softmax` and error
+            # in `torch.multinomial`.
+            _next_token_scores = torch.max(
```

Comment:
@ydshieh, softmax should be able to handle `-inf` correctly actually.
You can try:

```python
torch.nn.functional.softmax(torch.tensor([0, float("-inf")]))
```

which works as mathematically expected.

 It's only when **all** values are `-inf` that it doesn't work in which case this fix won't help because the generation is broken.